### PR TITLE
[test] Switch to C++ HTTP server

### PR DESCRIPTION
**Pull request by jmalanen**
Replace node.js based HTTP server used in test with C++ based implementation.

Fixes: mapbox/mapbox-gl-native-team#44

### CODE REVIEW

File: test/src/mbgl/test/http_server.cpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    server->Get(R"(/load/(\d+))", [](const Request req, Response& res) {
+        auto numbers = req.matches[1];
+        res.set_content("Request " + std::string(numbers), "text/plain");
+    });
+
+    server->listen("127.0.0.1", 3000);
+}
+
+HttpServer::HttpServer() {
+    server = std::make_unique<httplib::Server>();
```
Comment (alexshalamov) [inline test/src/mbgl/test/http_server.cpp]: Quickly checked httplib and it has `CPPHTTPLIB_THREAD_POOL_COUNT` option that might be useful to reduce number of threads?

File: next/platform/macos/macos.cmake [hunk 2]
```diff
@@ -165,24 +165,34 @@ add_executable(
...
 add_executable(
     mbgl-benchmark-runner
     ${MBGL_ROOT}/platform/default/src/mbgl/benchmark/main.cpp
 )
 
+target_include_directories(
+    mbgl-benchmark-runner
+    PUBLIC ${MBGL_ROOT}/benchmark/include
+)
+
```
Comment (alexshalamov) [inline next/platform/macos/macos.cmake]: Did we forget to add it in previous PRs?

File: next/platform/macos/macos.cmake [hunk 3]
```diff
@@ -165,24 +165,34 @@ add_executable(
...
 target_compile_definitions(
     mbgl-test-runner
     PRIVATE WORK_DIRECTORY=${MBGL_ROOT}
 )
 
 target_link_libraries(
     mbgl-test-runner
-    PRIVATE mbgl-compiler-options mbgl-test
+    PRIVATE mbgl-compiler-options -Wl,-force_load mbgl-test
```
Comment (alexshalamov) [inline next/platform/macos/macos.cmake]: Could you please tell why linker options were changed?

File: next/platform/macos/macos.cmake (hunk 2 above)
Comment (jmalanen) [reply next/platform/macos/macos.cmake]: Yes

File: next/platform/macos/macos.cmake (hunk 3 above)
Comment (jmalanen) [reply next/platform/macos/macos.cmake]: Benchmarks and (g)tests are now built as static libraries. They use a macro to instantiate the tests and if the static library is not linked with the force flag the linker will not include all test classes.

File: test/src/mbgl/test/http_server.cpp (hunk 1 above)
Comment (jmalanen) [reply test/src/mbgl/test/http_server.cpp]: Very good point, will limit the nbr of threads.

File: test/text/local_glyph_rasterizer.test.cpp [hunk 4]
```diff
@@ -52,7 +52,7 @@ class LocalGlyphRasterizerTest {
 } // end namespace
 
 // Enabling Qt requires adding a CJK-compatible font in the CI image.
-#if defined(__APPLE__)
+#if defined(__APPLE__) && !defined(__QT__)
```
[the remark this answers is not in the document]
Comment (jmalanen) [reply test/text/local_glyph_rasterizer.test.cpp]: Checked the result and it was ok, only the font had been changed. Updated the expected.png.

**Pull request merged**